pkg/ingester/ingester_v2.go

```diff
@@ -1280,3 +1312,435 @@ func (i *Ingester) v2FlushHandler(w http.ResponseWriter, _ *http.Request) {
 
 	w.WriteHeader(http.StatusNoContent)
 }
+
+// createBackfillTSDB creates a TSDB for a given userID and bucketName, and returns the created db.
+func (i *Ingester) createBackfillTSDB(userID, bucketName string, blockRange int64) (*userTSDB, error) {
```

Comment:
There too much code duplicated with `createTSDB()`. We should refactor `createTSDB()` to do things once.